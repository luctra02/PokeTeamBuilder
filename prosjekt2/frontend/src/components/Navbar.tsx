import { useState } from 'react';
import '../styles/Navbar.css';
import { Link, useNavigate } from 'react-router-dom';

function Navbar() {

  const searchStorage = sessionStorage.getItem('searchValue')
  const searchValue: string = searchStorage ? JSON.parse(searchStorage) : '';

  const [searchTerm, setSearchTerm] = useState(searchValue);
  const navigate = useNavigate();

  function clearSessionStorage(){
    sessionStorage.clear()
    setSearchTerm('')
  }
  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setSearchTerm(e.target.value);
  };

  const handleSearch = (e: React.KeyboardEvent<HTMLInputElement>) => {
    if (e.key === 'Enter') {
      sessionStorage.setItem("searchValue", JSON.stringify(searchTerm))
      sessionStorage.removeItem("type")
      navigate('/');
    }
  };

  return (
    <nav className="navbar">
      <figure className="navbar-logo">
        <Link to="/" onClick={clearSessionStorage}>
          PokemonTeamBuilder
        </Link>
      </figure>
      <div className="navbar-item nohover searchbar">
          <input
            type="text"
            placeholder="Search Pokemon Name..."
            value={searchTerm}
            onChange={handleInputChange}
            onKeyPress={handleSearch}
          />
        </div>
      <ul className="navbar-list">
        {/* <a>
        <li className="navbar-item noselect" onClick={toggleMode}>
          {mode === 'light' ? 'Dark Mode' : 'Light Mode'}
        </li>
        </a> */}
        <Link to="/" onClick={clearSessionStorage}>
          <li className="navbar-item">Home</li>
        </Link>
        <Link to="/team" onClick={clearSessionStorage}>
          <li className="navbar-item">My Team</li>
        </Link>
      </ul>
    </nav>
  );
}

export default Navbar;
