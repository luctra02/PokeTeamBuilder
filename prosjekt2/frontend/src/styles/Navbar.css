.navbar {
  position: absolute;
  top: 0;
  width: 100%;
  height: 70px;
  left: 0;
  display: flex;
  flex-direction: row;
  z-index: 100;
}

.navbar-logo {
  height: 50px;
  padding: 10px;
  display: flex;
}

.navbar-list {
  height: 50px;
  padding: 10px;
  right: 0;
  list-style: none;
  display: flex;
  justify-content: center;
  position: absolute;
  margin: 0.4rem 0.3rem;
}

.navbar-item {
  margin: 0 10px;
}

a {
  text-decoration: none;
  color: white;
}

.navbar-item a {
  text-decoration: none;
  color: white;
}

.navbar-item a:hover {
  text-decoration: underline;
}

.navbar-logo p {
  color: white;
  font-size: 1.5rem;
  margin: 0.4rem 0.3rem;
}

input[type='text'] {
  width: 100%;
  padding: 5px 10px;
  box-sizing: border-box;
  background-color: rgba(0, 0, 0, 0.2);
  color: white;
  border-radius: 15px;
  border: none;
}
