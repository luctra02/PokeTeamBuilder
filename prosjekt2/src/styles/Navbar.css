.navbar {
    position: absolute;
    top: 0;
    width: 100%;
    height: 70px;
    left: 0;
    display: flex;
    flex-direction: row;
  }
  
  .navbar-logo {
    height: 50px;
    padding: 10px;
    display: flex;
  }
  
  .navbar-list {
    height: 50px;
    padding: 10px;
    right: 0;
    list-style: none;
    display: flex;
    justify-content: center;
    position: absolute;
    margin: 0.4rem 0.3rem;
  }
  
  .navbar-item {
    margin: 0 10px;
  }
  
  a {
    text-decoration: none;
    color: white;
  }
  
  .navbar-item a {
    text-decoration: none;
    color: white;
  }
  
  .navbar-item a:hover {
    text-decoration: underline;
  }
  
  .navbar-logo p {
    color: white;
    font-size: 1.5rem;
    margin: 0.4rem 0.3rem;
  }
  
