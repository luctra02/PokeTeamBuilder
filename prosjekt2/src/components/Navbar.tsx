import "../styles/Navbar.css"
import { Link } from "react-router-dom";

function Navbar () {
    return (
        <nav className="navbar">
            <div className="navbar-logo">
                <Link to="/project2/">
                    <p>Pokemon</p>
                </Link>
            </div>

            <ul className="navbar-list">
                <Link to='/'><li className="navbar-item"><a href="/">Home</a></li></Link>
                <div className="navbar-item">
                    <input type="text" placeholder="Search" />
                </div>
            </ul>
            
        </nav>

        )
    }

export default Navbar