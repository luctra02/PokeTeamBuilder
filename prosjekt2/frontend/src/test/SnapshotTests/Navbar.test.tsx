import { screen } from '@testing-library/react';
import { allMocks as mocks} from '../mocks/mockQueries';
import { testPageRender } from '../testUtils';
import Navbar from '../../components/Navbar';




describe('renders a pokemon', () => {
    it('renders the correct ', async () => {
        const {asFragment} = testPageRender(<Navbar/>, { mocks });
        expect(screen.getByText(/Home/i)).toBeInTheDocument();
        expect(screen.getByText(/My Team/i)).toBeInTheDocument();

        expect(asFragment()).toMatchSnapshot();
    });
  }); 
