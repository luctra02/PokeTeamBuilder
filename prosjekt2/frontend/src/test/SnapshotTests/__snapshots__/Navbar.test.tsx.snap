// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders a pokemon > renders the correct  1`] = `
<DocumentFragment>
  <nav
    class="navbar"
  >
    <div
      class="navbar-logo"
    >
      <a
        href="/"
      >
        PokemonTeamBuilder
      </a>
    </div>
    <ul
      class="navbar-list"
    >
      <a>
        <li
          class="navbar-item noselect"
        >
          Dark Mode
        </li>
      </a>
      <a
        href="/"
      >
        <li
          class="navbar-item"
        >
          Home
        </li>
      </a>
      <a
        href="/team"
      >
        <li
          class="navbar-item"
        >
          My Team
        </li>
      </a>
      <div
        class="navbar-item nohover"
      >
        <input
          placeholder="Search Pokemon Name"
          type="text"
          value=""
        />
      </div>
    </ul>
  </nav>
</DocumentFragment>
`;
